Firebase/Core/FIRComponentContainer.m

```diff
@@ -161,14 +171,16 @@ - (nullable id)instanceForProtocol:(Protocol *)protocol {
 
 - (void)removeAllCachedInstances {
   // Loop through the cache and notify each instance that is a maintainer to clean up after itself.
-  for (id instance in self.cachedInstances.allValues) {
-    if ([instance conformsToProtocol:@protocol(FIRComponentLifecycleMaintainer)] &&
-        [instance respondsToSelector:@selector(appWillBeDeleted:)]) {
-      [instance appWillBeDeleted:self.app];
+  dispatch_sync(_containerQueue, ^{
+    for (id instance in self.cachedInstances.allValues) {
+      if ([instance conformsToProtocol:@protocol(FIRComponentLifecycleMaintainer)] &&
+          [instance respondsToSelector:@selector(appWillBeDeleted:)]) {
```

Comment:
Isn't the selector check redundant in a mandatory protocol?

Reply:
I think it's still good practice - I can't remember if it'd result in a compiler error or just a warning but if it's a warning, I'd rather skip cleanup than crash with unknown selector.